### PR TITLE
NXP-30226 more stats (master)

### PR DESCRIPTION
**Pull request by atchertchian**

### CODE REVIEW

File: modules/nuxeo-apidoc-core/src/main/java/org/nuxeo/apidoc/export/stats/AbstractJsonContributionStatsExporter.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    protected List<String> addAliases(DistributionSnapshot distribution, List<String> xps) {
+        List<String> res = new ArrayList<>();
+        xps.forEach(xp -> {
+            res.add(xp);
+            ExtensionPointInfo xpInfo = distribution.getExtensionPoint(xp);
+            if (xpInfo != null) {
+                res.addAll(xpInfo.getAliases());
+            }
+        });
+        return res;
```
Comment (efge) [inline modules/nuxeo-apidoc-core/src/main/java/org/nuxeo/apidoc/export/stats/AbstractJsonContributionStatsExporter.java:71]: I know it's just a renamed class but... If you had a method that takes a xp and returns a stream of the xp and all its aliases, you could use a simple `flatMap` and collector here.

**Review by efge — APPROVED** [review]

File: modules/nuxeo-apidoc-webengine/src/main/resources/skin/views/distribution/stats.ftl [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+<@block name="footer_scripts">
+  <script>
+    const rootURL = '${Root.path}';
+    const exporterName = 'jsonContributionStats';
+    const loadingDataText = "Loading data";
+    const noDataText = "No data";
+
+    const render = () => {
+      $("#feedback").html(loadingDataText);
+      $("#errors").html('');
```
Comment (nelsonsilva) [inline modules/nuxeo-apidoc-webengine/src/main/resources/skin/views/distribution/stats.ftl:81]: nit: there's a `.empty()`

File: modules/nuxeo-apidoc-webengine/src/main/resources/skin/views/distribution/stats.ftl [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+        $("#feedback").html(noDataText);
+        $("#codeTypeStats").html('');
+        $("#xpStats").html('');
+        $("#studioStats").html('');
+        $("#codeTypeStatsDetails").html('');
+        $("#xpStatsDetails").html('');
+        $("#studioStatsDetails").html('');
+      } else {
+        var allStats = [];
+        distribKeys.map(d => Plotly.d3.json(rootURL + '/' + d + '/export/' + exporterName, (err, data) => renderDistrib(err, data, d, total, allStats)));
```
Comment (nelsonsilva) [inline modules/nuxeo-apidoc-webengine/src/main/resources/skin/views/distribution/stats.ftl:96]: nit: I'd use template strings `${rootURL}/${d}....`

File: modules/nuxeo-apidoc-webengine/src/main/resources/skin/views/distribution/stats.ftl [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+      }
+    };
+
+    const renderDistrib = (error, rawData, distribKey, total, allStats) => {
+      var count = allStats.length + 1;
+      $("#feedback").html('Loading data ' + count + '/' + total);
```
Comment (nelsonsilva) [inline modules/nuxeo-apidoc-webengine/src/main/resources/skin/views/distribution/stats.ftl:102]: same nit: template strings

**Review by nelsonsilva — APPROVED** [review]

File: modules/nuxeo-apidoc-webengine/src/main/resources/skin/views/distribution/stats.ftl (hunk 3 above)
Comment (atchertchian) [reply modules/nuxeo-apidoc-webengine/src/main/resources/skin/views/distribution/stats.ftl:96]: Strangely that did not work ok (replacements were always empty) so I gave up trying to understand why

**Pull request merged**